File: totosuki/ABC/B/097.py
X = int(input())
se = {1}
for n in range(100):
  for m in range(2, 10):
    x = n ** m
    if x <= X:
      se.add(x)

print(max(se))

#import math
# X = int(input())

# print(int(math.sqrt(X)) ** 2)

File: OCH_Tao/ABC/B/097.js
const X = parseInt((require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n"))[0], 10);
let L = [];
for (let i = 2; i < 10; i++) {
  for (let j = 1; j < 32; j++) {
    L.push(j ** i);
  }
}
L = [...new Set(L)];
L = L.filter(a => a <= X);
console.log(Math.max(...L));